Dockerfile

```diff
@@ -1,9 +1,14 @@
-FROM debian:12 as builder
-ARG TARGETARCH
-RUN apt-get update && apt-get install -y curl clang gcc llvm make libbpf-dev -y
-RUN curl -LO https://go.dev/dl/go1.20.linux-${TARGETARCH}.tar.gz && tar -C /usr/local -xzf go*.linux-${TARGETARCH}.tar.gz
-ENV PATH="/usr/local/go/bin:${PATH}"
+FROM golang:1.21.3-bullseye as builder
+
 WORKDIR /app
+
+RUN apt-get update && apt-get install -y curl clang gcc llvm make libbpf-dev
+
+# pre-copy/cache go.mod for pre-downloading dependencies and only redownloading
+# them in subsequent builds if they change
+COPY go.mod go.sum ./
```

Comment:
I think we should also do the same for `internal/tools/go.mod` and `internal/tools/go.sum`.